### PR TITLE
Zoom Pipeline editor in/out/reset

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?
- [x] A zoom feature that zooms only in the study window, and doesn't just increase the size of everything on the page.

#### Bonus:
- [x] e2e: One more try to get rid of the lock on the jupyters study
- [x] bug fixed: "Delete" and "Esc" commands were always active

Zooming

Toolbar:
image

## Related issue number
closes https://github.com/ITISFoundation/osparc-issues/issues/253
realted to https://github.com/ITISFoundation/osparc-issues/issues/368

## Checklist

- [ ] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [ ] Unit tests for the changes exist
- [ ] Runs in the swarm
- [ ] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: services/web/client/source/class/osparc/component/workbench/WorkbenchUI.js [hunk 1]
```diff
@@ -288,6 +303,17 @@ qx.Class.define("osparc.component.workbench.WorkbenchUI", {
       this.__updateHint();
     },
 
+    __updateWorkbenchLayoutSize: function(position) {
```
Comment (KZzizzle) [inline services/web/client/source/class/osparc/component/workbench/WorkbenchUI.js:306]: nice

**Review by sanderegg — APPROVED** [review]
pretty cool!!
what happens if I unzoom, move away and lose my workbench? can I "zoom all"?

**Comment by GitHK** [comment]
This is OKish but I would ask for the following:
- if you are using a trackpad the zoom is way too fast (don't know how to fix), but a button to reset to the original coordinates would be good
- drag with center mouse button pressed.


**Image showing how hard it is to use with the trackpad.**  I was zooming super slowly.
hard_time_trackpad

**Comment by odeimaiz** [comment]
> This is OKish but I would ask for the following:
> 
> * if you are using a trackpad the zoom is way too fast (don't know how to fix), but a button to reset to the original coordinates would be good
> * drag with center mouse button pressed.
> 
> **Image showing how hard it is to use with the trackpad.** I was zooming super slowly.
> hard_time_trackpad

- I would need your trackpad to see how the event looks like or you PR the extension. Adding the zoom in/out, reset to original and zoom all buttons is also an option.

- If you want to move around the workbench you have the scroll bar. If you also want that center mouse button feature, we can implement it in the next iteration, create an issue, please.

**Comment by odeimaiz** [comment]
> pretty cool!!
> what happens if I unzoom, move away and lose my workbench? can I "zoom all"?

When zooming in and out, the position of the nodes is taken into account, so I don't think you can actually lose the workbench. I will implement the "zoom all" button.

**Comment by GitHK** [comment]
Added issue to keep track of dragging https://github.com/ITISFoundation/osparc-simcore/issues/2016

**Review by ignapas — COMMENTED** [review]
This is super cool.
I just saw that the unlink button disappears when you zoom in, and that the scroll bars act weird when you zoom out. If you zoom out and everything is visible, there shouldn't be any scroll bar right? In fact, they don't do anything.
I think it would be cool to implement the suggested features for this first iteration:
* zoom in and zoom out buttons for users without a mouse wheel
* zoom all that adjusts zoom to have the whole workbench in the view

Something like in GoogleMaps (the button at the top would be the 'zoom all' feature). And the unlink button would appear/disappear on top, too.
Screenshot 2020-12-03 at 11 54 11

**Comment by KZzizzle** [comment]
> This is super cool.
> I just saw that the unlink button disappears when you zoom in, and that the scroll bars act weird when you zoom out. If you zoom out and everything is visible, there shouldn't be any scroll bar right? In fact, they don't do anything.
> I think it would be cool to implement the suggested features for this first iteration:
> 
> * zoom in and zoom out buttons for users without a mouse wheel
> * zoom all that adjusts zoom to have the whole workbench in the view
> 
> Something like in GoogleMaps (the button at the top would be the 'zoom all' feature). And the unlink button would appear/disappear on top, too.
> Screenshot 2020-12-03 at 11 54 11

Agreed with this - it would be super ultra cool.

Also enthusiastically second the "reset to original" option mentioned by Andrei. 

I haven't managed to run this on my computer yet and so haven't really "tested". If you think it makes sense, I would also make the zoom ability triggered with holding down  ```shift``` since that is the behavior of a lot of other apps. Users may try to scroll (even though we don't have that implemented yet) and it could be distracting if the thing zooms. 

Anyway AWESOME ODEI!!!

**Comment by KZzizzle** [comment]
I noticed something in zoomed in view and perhaps this is a separate issue - if you try to create a new service dragging from the out port of a service, the catalog will appear outside of the viewable space (in the extreme case you cannot see it at all). You tell me if you want me to put this as a separate issue =) 

image

**Comment by odeimaiz** [comment]
> I haven't managed to run this on my computer yet and so haven't really "tested". If you think it makes sense, I would also make the zoom ability triggered with holding down `shift` since that is the behavior of a lot of other apps. Users may try to scroll (even though we don't have that implemented yet) and it could be distracting if the thing zooms. Additionally I see that you now have scroll bars when you are zoomed in. If you have the `shift` implemented, then scrolling can be left to the intuitive function which is scrolling down within the workbench window
> 
> Anyway AWESOME ODEI!!!

That one could be addressed here https://github.com/ITISFoundation/osparc-simcore/issues/2016

**Comment by odeimaiz** [comment]
> I noticed something in zoomed in view and perhaps this is a separate issue - if you try to create a new service dragging from the out port of a service, the catalog will appear outside of the viewable space (in the extreme case you cannot see it at all). You tell me if you want me to put this as a separate issue =)
> 
> image

That has always been the case, I can try forcing the catalog to be always completely visible.

**Review by KZzizzle — APPROVED** [review]
Super cool!

**Pull request merged**